clients/rospy/src/rospy/impl/tcpros_pubsub.py

```diff
@@ -387,6 +387,10 @@ def __init__(self, connection, queue_size):
 
         self._thread = threading.Thread(target=self._run)
         self._thread.start()
+        self._connection.set_cleanup_callback(self._closed_connection_callback)
```

Comment:
Should this line be moved up? At least above the thread start - likely directly under the creation of the condition.